experimental/torch_xla2/examples/train_llama_torchtitan/train_llama.py

```diff
@@ -0,0 +1,283 @@
+import os
+import time
+import logging
+from typing import Tuple
+from collections import defaultdict
+import functools
+
+import torch
+import torch.nn.functional
+from torch.utils import _pytree as pytree
+
+import torch_xla2
+import torch_xla2.interop
+from torch_xla2.interop import jax_view, torch_view, JittableModule
+import jax
+import jax.numpy as jnp
+from jax.experimental import shard_map
+from jax.experimental import mesh_utils
+import optax
+
+from torchtitan.models.llama import llama3_configs
+from torchtitan.models.llama import model as titan
+
+P = jax.sharding.PartitionSpec
+
+def _setup_default_env():
+  os.environ.setdefault('TF_CPP_MIN_LOG_LEVEL', '1')
+  os.environ.setdefault('GRPC_VERBOSITY', 'ERROR')
+  os.environ.setdefault('ALLOW_MULTIPLE_LIBTPU_LOAD', '1')
+  # only need for tpu v4
+  # os.environ.setdefault('TPU_MEGACORE', 'megacore_dense')
```

Comment:
Maybe it's alright to always set `TPU_MEGACORE`?

Reply:
it crashes if not on v4 :(

Reply:
okie ><